mithril-aggregator/src/database/provider/certificate.rs

```diff
@@ -628,9 +629,18 @@ impl CertificateRetriever for CertificateRepository {
     ) -> Result<Certificate, CertificateRetrieverError> {
         self.get_certificate(certificate_hash)
             .await
-            .map_err(|e| CertificateRetrieverError::General(e.to_string()))?
-            .ok_or(CertificateRetrieverError::General(
-                "certificate does not exist".to_string(),
+            .map_err(|e| {
+                CertificateRetrieverError::new(
+                    &format!(
+                        "Certificate repository can not retrieve certificate: '{}'",
+                        certificate_hash
+                    ),
+                    anyhow!(e),
+                )
+            })?
+            .ok_or(CertificateRetrieverError::new(
+                &format!("Certificate does not exist: '{}'", certificate_hash),
+                anyhow!("Certificate does not exist"),
```

Comment:
Maybe a simpler approach to this error wrapper would be to create a type that just wrap an anyhow error and add contexts to the inner error. This way it is easier to iterate over the error's different contexts IMHO.

```rust
type MyError(AnyhowError);

let a = that_fails().map_err(|e| MyError(e.with_context("my error message"))?;
```